app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -7,6 +7,26 @@ def run
           results = SerializableArray.new(results)
           SearchTaskDefinition.create_output(results, output)
         end
+
+        def self.dereference_output(storage, path)
```

Comment:
1. ack
2. ack - it's the app developer's responsibility to use sensible keys
3. dynflow extracts the dependencies in after planning: we are using the fact, that the output is not computed in planning, so the output object there is just a stub that is used for recording the dependencies: if you tried to use `action.output[:one][:two][three]` in planning, it would give you just an output reference, and this one is replaces by specific data when the run method of the receiver starts (after the sender's run is evaluated and real output is filled in)
4. ack

It seems we are in good agreement on the expectations, I hope to get soon to a draft implementation of this in Dynflow, doesn't look that hard right now (the devil is in details)